include/inexor/vulkan-renderer/world/cube.hpp

```diff
@@ -92,10 +92,12 @@ class Cube : public std::enable_shared_from_this<Cube> {
     Cube(Cube &&rhs) noexcept;
     ~Cube() = default;
     Cube &operator=(Cube rhs);
+    Cube &operator=(Cube &&) = delete;
+
     /// Get child.
     std::shared_ptr<Cube> operator[](std::size_t idx);
     /// Get child.
-    const std::shared_ptr<const Cube> operator[](std::size_t idx) const;
+    const Cube &operator[](std::size_t idx) const;
```

Comment:
I will search through the engine again to specifically search for smart pointers which can be replaced with const references in a separate pull request.

Reply:
I would avoid this, especially because they are saved as `shared_ptr`, you can still get a const reference later, also you will loose the safety that the cube you access will exist. Aiming at multi threading.